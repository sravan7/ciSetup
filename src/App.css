.App {
  display: flex;
  width: 100%;
  height:  calc(100% - 10px);
  flex-flow: row wrap;
  justify-content: center;
  align-items: center;
}

.App-header{
      width: 100%;
      height: 74px;
      background-color: #61dafb;
  }
  .App-login{
    width: 70%;
    height: 50%;
    background-color: white;
    display: flex;
    flex-flow: column wrap;
    justify-content: center;
    border: 1px solid #4caf50;
    border-radius: 4px;
    box-shadow: 0px 0px 5px 0px rgba(0, 100, 0, 0.1), 0px 0px 3px 0px rgba(0, 100, 0, 0.3) ;
    align-items: center;
  }
  .App-login>form{
    width: 70%;
    background-color: white;
    display: flex;
    flex-flow: column wrap;
    justify-content: center;
    align-items: center;
  }
  .login-header{
    width: 70%;
    margin-bottom: 5%;
    text-align: center;
    font-weight: 500;
    font-size: large;
  }
  .login-error{
    width: 70%;
    margin-bottom: 2%;
    text-align: center;
    color: #ff0000;
    font-size: medium;
  }
  .login-email{
      
  }
  .loginControls{
    width: 70%;
    margin-bottom: 5%;
  }
  .custom-button{
    width: 70%;
    height: 40px;
  }
  
.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

/* .App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
} */

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

/********************************* Header styles *********************************/
.app-bar{
  top: 0;
  left: 0;
  width: 100%;
  height: 10%;
  position: fixed;
  z-index: 1;
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  border-bottom: 1 px solid rgba(0, 0,0,0.3);
  box-shadow: 2px 0 3px 0px rgba(0,0,0,0.5);
}
.menu-icon{
  /* margin-left: 4%; */
  width: 32px;
  height: 32px;
  color: green;
  background-image: url('/menu-24px.svg');
  background-repeat: no-repeat;
  background-size: contain;
  cursor: pointer;
}
#menuIcon{
  height: 30px;
  width: 100%;
  outline: none;
  border: none;
  text-align: left;
  box-sizing: border-box;
  border-bottom: 1px solid rgba(0, 0,0,0.5);
  cursor: pointer;
}
.app-search{
  /* flex: 1 1; */
  width: 60%;
  height: 32px;
  margin: 0;
  padding: 0;
}
.inbox-count{
  cursor: pointer;
  width: 32px;
  height: 32px;
  color: green;
  background-image: url('/mail-24px.svg');
  background-repeat: no-repeat;
  background-size: contain;
}
.notification{
  width: 32px;
  height: 32px;
  color: green;
  background-image: url('/notifications-24px.svg');
  background-repeat: no-repeat;
  background-size: contain;
  cursor: pointer;
}
.app-logout{
  width: 32px;
  height: 32px;
  color: green;
  background-image: url('/exit_to_app-24px.svg');
  background-repeat: no-repeat;
  background-size: contain;
  cursor: pointer;
}
/*****************************************  extrasSideMenu ***************************************/
.extras-side-menu{
 position: absolute;
 width: 10%;
 height: 100%;
 z-index: 2;
 background-color: rgba(0, 0,0, 0.4);
 top: 0;
 left: -10%;
 transition: all 0.2s;
}
.side-menu-push{
  left: 0;
}
.app-bar-shrink{
  left: 10%;
  width: 90%;
}


/***************************************** App Content *************************************************/
.app-content{
  width: 100%;
  height: 89%;
  top: 10%;
  left: 0;
  display: flex;
  position: absolute;
  background-color: rgba(255, 255,255, 0.7);
}
.app-content-shrink{
  left: 10%;
  width: 90%;
  background-color: rgba(85, 71, 71, 0.5);
}
/****************************************** menu section ***********************************************/
.menu{ 
  margin-top: 3%;
  height: 95%;
  width: 25%;
  background-color: rgba(85, 71, 71, 0.5);
}
.menu-content{
  width: 90%;
  height: 100%;
  margin: 0 5% 0 5%;
  display: flex;
  flex-flow:column;
  justify-content: space-evenly;
  align-items: flex-start;
  background-color: rgba(0, 0,0, 0.4);
}
.mail-compose-button{
  width: 100%;
  height: 7%;
  font-size: 16px;
  background-color: rgba(0,180,148,1);
  border-radius: 4px;
  outline: none;
  border: none;
  color: white;
  box-shadow: 0 0 3px rgba(0,180,148,0.5), 0 0 0 2px rgba(0,180,148,0.1);
}
.menu-folders{
  height: 32%;
}
.menu-categories{
  height: 32%;
}
.menu-labels{
  height: 10%;
}





/************************************************** app-section ***************************************/
.app-section{
  margin-top: 3%;
  height: 95%;
  background-color: yellow;
  width: 72%;
}
.mail-header{
  width: 100%;
  height: 20%;
  background-color: green;
}
.mail-list{
  height: 100%;
  margin: 0;
  padding: 0;
  list-style-type: none;
}
.mail-item{
  width: 100%;
  height: 6%;
  background-color: white;
}