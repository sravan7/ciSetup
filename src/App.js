import React, { useState } from 'react';
import logo from './logo.svg';
import './App.css';
import Login from './Login';
import Dashboard from "./Dashboard"

function App() {
  let initialData = { isLogin: false, value: "" };
  let accessToken = window.localStorage.getItem("accessToken")
  if (accessToken) {
    initialData = { isLogin: true, value: accessToken };
  }
  const [loginState, setLoginState] = useState(initialData);
  const handleLogin = (result) => {
    window.localStorage.setItem("accessToken", result.value)
    console.log(result)
    setLoginState({ isLogin: true, value: result.value })
  }
  return (
    <div className="App">
      {/* <div className="App-header"></div>  */}
      {!loginState.isLogin ?
        <Login handleLogin={handleLogin} /> :
        <Dashboard mail={loginState.value} />
      }
    </div>
  );
}

export default App;
