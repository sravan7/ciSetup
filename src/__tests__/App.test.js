import React from 'react';
import { render } from '@testing-library/react';
import App from '../App.js';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

const email = "sravan@email.com";
const password = "1234"
test('login view ', () => {
  const output = mount(<App />);
  output.find(".login-email").value=email;
  output.find(".login-password").value=password;
  output.find("button").simulate("click")
});
// test('snapshot test', () => {
//   const tree = renderer.create(<App />).toJSON();
//     expect(tree).toMatchSnapshot();
// });
