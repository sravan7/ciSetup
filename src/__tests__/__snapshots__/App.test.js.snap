// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="App"
>
  <div
    className="App-login"
  >
    <form
      action="none"
      onSubmit={[Function]}
    >
      <div
        className="login-header"
      >
        Sign In
      </div>
      <div
        className="loginControls"
      >
        <div
          className="MuiFormControl-root-2 MuiTextField-root-1 MuiFormControl-fullWidth-5"
        >
          <label
            className="MuiFormLabel-root-18 MuiInputLabel-root-6 MuiInputLabel-formControl-12 MuiInputLabel-animated-15 MuiInputLabel-outlined-17 Mui-required Mui-required"
            data-shrink={false}
            htmlFor="login-email"
            id="login-email-label"
          >
            email
            <span
              className="MuiFormLabel-asterisk-25 MuiInputLabel-asterisk-11"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root-41 MuiOutlinedInput-root-26 MuiInputBase-fullWidth-51 MuiInputBase-formControl-42"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={true}
              className="MuiInputBase-input-52 MuiOutlinedInput-input-36"
              disabled={false}
              id="login-email"
              name="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="email"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-59 MuiOutlinedInput-notchedOutline-35"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-61"
              >
                <span>
                  email
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="loginControls"
      >
        <div
          className="MuiFormControl-root-64 MuiTextField-root-63 MuiFormControl-fullWidth-67"
        >
          <label
            className="MuiFormLabel-root-80 MuiInputLabel-root-68 MuiInputLabel-formControl-74 MuiInputLabel-animated-77 MuiInputLabel-outlined-79 Mui-required Mui-required"
            data-shrink={false}
            htmlFor="login-password"
            id="login-password-label"
          >
            password
            <span
              className="MuiFormLabel-asterisk-87 MuiInputLabel-asterisk-73"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root-103 MuiOutlinedInput-root-88 MuiInputBase-fullWidth-113 MuiInputBase-formControl-104"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input-114 MuiOutlinedInput-input-98"
              disabled={false}
              id="login-password"
              name="password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="password"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-121 MuiOutlinedInput-notchedOutline-97"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-123"
              >
                <span>
                  password
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="loginControls"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained custom-button MuiButton-containedSecondary MuiButton-fullWidth"
          disabled={false}
          id="cutom-button"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
          value="Sign In"
        >
          <span
            className="MuiButton-label"
          >
            submit
          </span>
        </button>
      </div>
    </form>
  </div>
</div>
`;
